packages/visual-editing/src/ui/ElementOverlay.tsx

```diff
@@ -299,15 +327,117 @@ export const ElementOverlay = memo(function ElementOverlay(props: ElementOverlay
     scrolledIntoViewRef.current = focused === true
   }, [focused, wasMaybeCollapsed, enableScrollIntoView])
 
+  function getTimeDifferenceInDays(targetDateStr: string): number {
+    const now = new Date()
+
+    const targetDate = new Date(targetDateStr)
+    const timeDifference = targetDate.getTime() - now.getTime()
+    // Convert milliseconds to days
+    return Math.round(timeDifference / (1000 * 60 * 60 * 24))
+  }
+
+  let nearestUpcomingRelease = null
+  let timeTilNextRelease = null
+
+  const showReleasePreview = element.getAttribute('data-sanity-release-preview') === 'true'
+
+  if (showReleasePreview) {
+    const releaseVersions = versions ? versions.filter((v) => v._id.startsWith('versions.')) : []
+    const overlayDocumentReleases = releaseVersions.map((v) => {
+      const releaseName = v._id.split('.').at(1)
+
+      return releases.find((r) => r.name === releaseName)
+    })
+
+    if (overlayDocumentReleases.length > 0) {
+      const asapReleases = overlayDocumentReleases.filter((r) => r.metadata.releaseType === 'asap')
+
+      const scheduledReleases = overlayDocumentReleases.filter(
+        (r) => r.metadata.releaseType === 'scheduled',
+      )
+
+      if (asapReleases.length > 0) {
+        const asapReleasesSorted = asapReleases.sort(
+          (a, b) => new Date(a._createdAt).getTime() - new Date(b._createdAt).getTime(),
+        )
+
+        nearestUpcomingRelease = asapReleasesSorted[0]
+      } else if (scheduledReleases.length > 0) {
+        const scheduledReleasesSorted = scheduledReleases.sort(
+          (a, b) =>
+            new Date(a.metadata.intendedPublishAt).getTime() -
+            new Date(b.metadata.intendedPublishAt).getTime(),
+        )
+
+        nearestUpcomingRelease = scheduledReleasesSorted[0]
+
+        timeTilNextRelease = getTimeDifferenceInDays(
+          nearestUpcomingRelease.metadata.intendedPublishAt,
+        )
```

Comment:
for sorting we also need to consider the release.publishAt date, which will be added once the release is scheduled. 
So 
```
   const scheduledReleasesSorted = scheduledReleases.sort(
          (a, b) =>
            new Date(a.publishAt || a.metadata.intendedPublishAt).getTime() -
            new Date(b.publishAt || b.metadata.intendedPublishAt).getTime(),
        )
```

And we could also have releases that are neither asap or scheduled, which are the `undecided`

Maybe it's easier for now to copy this from `sanity` to use the same sort mechanism. https://github.com/sanity-io/sanity/blob/corel/packages/sanity/src/core/releases/hooks/utils.ts#L6-L49

Reply:
Also, if you are using releases from ` useReleases` you can trust that they will be sorted, so you could use the first match found.

Reply:
I am using `useReleases` now so all sorting logic is removed. Great call out!